### DISCUSSION TITLE
PnP and chunking causing issues on the SharePoint side of things

### DISCUSSION THREAD

**Discussion by chevol** [pnp/pnpcore#1196] [Q&A] [answered]
We have a strange issue where files less than 10mb are uploaded to SharePoint fine without any issues. However when a file is larger than 10mb and chunking happens then we receive an error. Is this a SharePoint setting that we need to adjust? Is there a way to turn off chunking in PnP? We are using an S3 upload to trigger our lambda function that uses PnP to upload a converted file to Sharepoint (not sure if that matters). I assume its something on the Sharepoint side of things because files less than 10mb work perfectly fine. Has anyone else experienced this before?

The error I get is "fail	Specified method is not supported."

**Comment by jansenbe** [comment]
@chevol : have not heard about this before. Does it repro on all your tenants?

**Reply by chevol** [reply]
@jansenbe Thanks for the reply, we use this internally so only have one tenant. The error suggests that its on the SharePoint side of things and only happens when PnP switches to chunking (which is odd). Any help would be greatly appreciated, thanks again for the reply and I hope you have a wonderful day!

**Reply by jansenbe** [reply]
@chevol : are you in a regular cloud or special cloud environment? I've just ran our test case against my demo tenant (https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core.Test/SharePoint/FilesTests.cs#L1948-L1966) and that's working fine. Did this ever work and suddenly started failing or did this never work? Can you try to build an isolated repro (like the test case for example) and see if the issue still happens.

**Reply by chevol** [reply]
@jansenbe we are in a regular cloud environment to my knowledge. We are writing from a ResponseStream as the file is in S3, not sure if that matters. This has never worked for us unfortunately so its not a recent failure.  Let me toss some code into a repo so you can see how we are calling it and maybe catch something we are missing. Thanks again for your response, we really appreciate it!

**Reply by chevol** [reply]
@jansenbe Here is an isolated repo:
https://github.com/chevol/PnpExample/blob/main/PNPSharepointUploader.cs
This works for files that are less then 10mb but anything over *when automatic chunking happens) gives us the "Specified method is not supported" error. Thanks again for you help!

**Reply by jansenbe** [reply]
@chevol : thanks for sharing the sample, this clarified you're not using PnP Core SDK but PnP Framework instead. A number of things you can try are:
- Replace the method `UploadFileAsync` with the actual code used in PnP Framework (https://github.com/pnp/pnpframework/blob/dev/src/lib/PnP.Framework/Extensions/FileFolderExtensions.cs#L1459-L1610) and see if that get's you any further. Possibly you can skip the chunked upload all together if that's possible
- Do interop with PnP Core SDK and try our upload:
  - Get a `PnPContext` from the CSOM `ClientContext`:  https://pnp.github.io/pnpcore/using-the-sdk/basics-pnpframework.html#using-pnp-core-sdk-when-pnp-framework-was-already-configured
  - Load the folder you want to upload to and then perform the upload: https://pnp.github.io/pnpcore/using-the-sdk/files-large.html#uploading-large-files

**Reply by chevol** [reply]
Thanks @jansenbe I will give that a try and report back so if others experience the same issue they can use this thread as a reference.

**Reply by chevol** [reply, marked as answer]
@jansenbe I took the first route and replaced the method UploadFileAsync with the actual code used in PnP Framework. Its now working fine and I am not too concerned with chunking since we are going from AWS to SharePoint. Thanks again for the help, maybe a future update could add an option/config to turn chunking off ;)
